crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -424,13 +425,22 @@ impl UserIdentityData {
 ///
 /// The user can explicitly pin the new identity to allow for legitimate
 /// identity changes (for example, in case of key material or device loss).
+///
+/// As soon as the cryptographic identity is verified (i.e. signed by our own
+/// trusted identity), a latch is set to remember it (`verified_latch`). Future
+/// interactions will expect this user to stay verified, in case of violation
+/// the user should be notified with a blocking warning.
 #[derive(Debug, Clone, Deserialize, Serialize)]
 #[serde(try_from = "OtherUserIdentityDataSerializer", into = "OtherUserIdentityDataSerializer")]
 pub struct OtherUserIdentityData {
     user_id: OwnedUserId,
     pub(crate) master_key: Arc<MasterPubkey>,
     self_signing_key: Arc<SelfSigningPubkey>,
     pinned_master_key: Arc<RwLock<MasterPubkey>>,
+    /// This tracks whether we've ever verified this user with any identity at
+    /// any point in time on this device. To use it in the future to detect
```

Comment:
"at any point in time" ... won't this get reset back to `false` when we "revoke verification", so actually this is not at any point in time - it's more like "we currently think they are verified", right?

Reply:
Yes it will revert only if you call `withdraw_verification`. I will clarify. 
I'll try to add also that it's "local", i.e not shared across devices

Reply:
Updated